### PR TITLE
Edit_team_asset

### PR DESCRIPTION
**Pull request by shmera6**

### CODE REVIEW

File: src/main/java/Domain/Game/Stadium.java [hunk 1]
```diff
@@ -1,4 +1,150 @@
...
+    private String stadName;
+    private String stadLocation;
+    private List<Team> homeTeams;
+
+    public final String namePropertyString = "Name";
+
+
+    public Stadium(String location)
+    {
+        homeTeams = new ArrayList<>();
```
Comment (galrosenthal) [inline src/main/java/Domain/Game/Stadium.java:22]: add to EntityManager or remove this constructor

File: src/main/java/Domain/Game/Team.java [hunk 2]
```diff
@@ -15,27 +20,41 @@
...
         seasons = new ArrayList<>();
     }
+
     /**
      * A method that returns a string containing all team information for display
+     *
      * @return string containing all team information.
      */
-    public String DisplayTeam(){
+    public String DisplayTeam() {
```
Comment (galrosenthal) [inline src/main/java/Domain/Game/Team.java:55]: remove this function

File: src/main/java/Domain/Users/TeamManager.java [hunk 3]
```diff
@@ -1,10 +1,174 @@
...
+    {
+        return false;
+    }
+
+    private boolean removePermission(Team team , TeamManagerPermissions permission)
+    {
+        return false;
+    }
+
+    private List<TeamManagerPermissions> allPermissionsPerTeam(Team team)
```
Comment (galrosenthal) [inline src/main/java/Domain/Users/TeamManager.java:53]: change name to get...

File: src/main/java/Domain/Users/TeamManager.java [hunk 4]
```diff
@@ -1,10 +1,174 @@
...
+    {
+        List<TeamManagerPermissions> permissions = new ArrayList<>();
+        if(this.permissionsPerTeam.get(team) != null)
+        {
+            permissions = this.permissionsPerTeam.get(team);
+        }
+        return permissions;
+    }
+
+    //todo: add permissions!
```
Comment (galrosenthal) [inline src/main/java/Domain/Users/TeamManager.java:63]: delete it!!!!!!

File: src/test/java/Acceptance/AcceptanceTests.java [hunk 5]
```diff
@@ -201,4 +205,93 @@ public void tearDown() throws Exception {
...
+
+        Unregistered abcUnreg = new Unregistered();
+        SystemUser abc = abcUnreg.login("abc1", "abc12345");
+        assertEquals(abc, abcCreate);
+
+        SystemUser elivyCreate = new SystemUser("elivy", "abc12345", "elisha levy");
+        Player playerElisha = new Player(elivyCreate, new Date());
+        beitShean.addTeamPlayer(abcOwner , playerElisha);
+        UIController.setSelector(6139);
+        assertTrue(Controller.modifyTeamAssetDetails(abc));
```
Comment (galrosenthal) [inline src/test/java/Acceptance/AcceptanceTests.java:292]: i think we should check fail alternative case

File: src/test/java/Domain/Game/TeamTest.java [hunk 6]
```diff
@@ -1,57 +1,702 @@
...
+        Coach coach = new CoachStub(new SystemUser("teamTest3", "gal"));
+        team.addTeamCoach(teamOwner, coach);
+        assertTrue(team.getAllAssets().size() == 3);
+        Coach secondCoach = new CoachStub(new SystemUser("teamTest4", "gal"));
+        team.addTeamCoach(teamOwner, coach);
+        assertTrue(team.getAllAssets().size() == 4);
+    }
+    /*Stadium*/
+    @Test
+    public void getAllAssetsITest_1() {
```
Comment (galrosenthal) [inline src/test/java/Domain/Game/TeamTest.java:499]: put the number befoer the ITest

File: src/test/java/Domain/Game/TeamTest.java [hunk 7]
```diff
@@ -1,57 +1,702 @@
...
+        assertTrue(team.getAllAssets().size() == 3);
+        Coach secondCoach = new CoachStub(new SystemUser("teamTest4", "gal"));
+        team.addTeamCoach(teamOwner, secondCoach);
+        assertTrue(team.getAllAssets().size() == 4);
+    }
+
+    /*Stadium , Player*/
+    @Test
+    public void getAllAssetsITest_2() {
+        Team team = new Team();
```
Comment (galrosenthal) [inline src/test/java/Domain/Game/TeamTest.java:521]: also here

File: src/test/java/Domain/Game/TeamTest.java [hunk 8]
```diff
@@ -1,57 +1,702 @@
...
+        team.addTeamCoach(teamOwner, coach);
+        assertTrue(team.getAllAssets().size() == 3);
+        Coach secondCoach = new CoachStub(new SystemUser("teamTest4", "gal"));
+        team.addTeamCoach(teamOwner, secondCoach);
+        assertTrue(team.getAllAssets().size() == 4);
     }
 
+    /*Stadium , Player , Coach*/
+    @Test
+    public void getAllAssetsITest_3() {
```
Comment (galrosenthal) [inline src/test/java/Domain/Game/TeamTest.java:541]: also here

File: src/test/java/Domain/Game/TeamTest.java [hunk 9]
```diff
@@ -1,57 +1,702 @@
...
+        team.addTeamCoach(teamOwner, coach);
+        assertTrue(team.getAllAssets().size() == 3);
+        Coach secondCoach = new Coach(new SystemUser("teamTest4", "gal"));
+        team.addTeamCoach(teamOwner, secondCoach);
+        assertTrue(team.getAllAssets().size() == 4);
+    }
 
+    /*Stadium , Player , Coach , TeamManager*/
+    @Test
+    public void getAllAssetsITest_4() {
```
Comment (galrosenthal) [inline src/test/java/Domain/Game/TeamTest.java:562]: also here

File: src/test/java/Service/ControllerTest.java [hunk 10]
```diff
@@ -32,23 +40,131 @@ public void addTeamOwnerUTest() {
...
+        Team team = new Team();
+        team.getTeamOwners().add(to);
+        assertTrue(to.addTeamToOwn(team));
+        UIController.setSelector(61115);
+        assertTrue(Controller.addAsset(test));
+    }
+
+
+    @Test
+    public void modifyTeamAssetDetailsUTest1() throws Exception {
```
Comment (galrosenthal) [inline src/test/java/Service/ControllerTest.java:133]: number after Test

Comment (shmera6) [reply src/test/java/Service/ControllerTest.java:133]: done

File: src/test/java/Domain/Game/TeamTest.java (hunk 9 above)
Comment (shmera6) [reply src/test/java/Domain/Game/TeamTest.java:562]: done

File: src/test/java/Domain/Game/TeamTest.java (hunk 8 above)
Comment (shmera6) [reply src/test/java/Domain/Game/TeamTest.java:541]: done

File: src/test/java/Domain/Game/TeamTest.java (hunk 7 above)
Comment (shmera6) [reply src/test/java/Domain/Game/TeamTest.java:521]: done

File: src/test/java/Domain/Game/TeamTest.java (hunk 6 above)
Comment (shmera6) [reply src/test/java/Domain/Game/TeamTest.java:499]: done

File: src/main/java/Domain/Users/TeamManager.java (hunk 4 above)
Comment (shmera6) [reply src/main/java/Domain/Users/TeamManager.java:63]: done

File: src/main/java/Domain/Users/TeamManager.java (hunk 3 above)
Comment (shmera6) [reply src/main/java/Domain/Users/TeamManager.java:53]: done

File: src/main/java/Domain/Game/Team.java (hunk 2 above)
Comment (shmera6) [reply src/main/java/Domain/Game/Team.java:55]: done

File: src/main/java/Domain/Game/Stadium.java (hunk 1 above)
Comment (shmera6) [reply src/main/java/Domain/Game/Stadium.java:22]: deleted!

File: src/main/java/Domain/Controllers/TeamController.java [hunk 11]
```diff
@@ -56,21 +58,275 @@ public static boolean addTeamOwner(String username, Team teamToOwn, TeamOwner ow
...
+    }
+
+    /**
+     * In case the user wants to remove Property to property list- choose which new property he wants to change.
+     * - assumption- list of Enums property
+     *
+     * @param allEnumValues
+     * @return propertyNewValueIndex -int-
+     */
+    private static int removePropertyToListProperty(List<Enum> allEnumValues) {
```
[the remark this answers is not in the document]
Comment (shmera6) [reply src/main/java/Domain/Controllers/TeamController.java]: Done

**Pull request merged**